File: 01_03_2025_Apply Operations to an Array.py
class Solution(object):
    def applyOperations(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        count = 0
        for i in range(len(nums) - 1):
            if nums[i] == nums[i + 1]:
                nums[i] *= 2
                nums[i + 1] = 0
            if nums[i] == 0:
                count += 1
        
        if nums[-1] == 0:
            count += 1

        ans = [num for num in nums if num != 0]
        ans.extend([0] * count)  # Append zeros at the end
        
        return ans

File: 01_03_2025_Apply Operations to an Array.cpp
class Solution {
    public:
        vector<int> applyOperations(vector<int>& nums) {
            int count=0;
            for(int i=0;i<nums.size()-1;i++){
                if(nums[i]==nums[i+1]){
                    nums[i]=nums[i]*2;
                    nums[i+1]=0;
                }
                if(nums[i]==0){
                    count++;
                }
            }
            if(nums[nums.size()-1]==0){
                count++;
            }
            vector<int> ans;
            for(int i=0;i<nums.size();i++){
                if(nums[i]!=0){
                    ans.push_back(nums[i]);
                }
            }
            while(count!=0){
                ans.push_back(0);
                count--;
            }
            return ans;
        }
    };